src/joystick/windows/SDL_dinputjoystick.c

```diff
@@ -1121,6 +1146,9 @@ SDL_DINPUT_JoystickUpdate(SDL_Joystick * joystick)
 void
 SDL_DINPUT_JoystickClose(SDL_Joystick * joystick)
 {
+    if (dinput == NULL) {
```

Comment:
This shouldn't be called if dinput is NULL. If that's not the case, then that's a bug in the higher level code and should be fixed.